pco/src/standalone/decompressor.rs

```diff
@@ -165,6 +165,26 @@ impl FileDecompressor {
     };
     Ok(MaybeChunkDecompressor::Some(res))
   }
+
+  /// Takes in compressed bytes (after the header, at the start of the chunks)
+  /// and returns a vector of numbers.
+  ///
+  /// Will return an error if there are any compatibility, corruption,
+  /// or insufficient data issues.
+  ///
+  /// This function exists (in addition to the [standalone
+  /// functions][crate::standalone]) because the user may want to peek at the
+  /// dtype, allowing them to know which type `<T>` to use here. There is no
+  /// analagous file compressor method because the user always knows the dtype
+  /// during compression.
+  pub fn simple_decompress<T: Number>(&self, mut src: &[u8]) -> PcoResult<Vec<T>> {
+    let mut res = Vec::with_capacity(self.n_hint());
+    while let MaybeChunkDecompressor::Some(mut chunk_decompressor) = self.chunk_decompressor(src)? {
+      chunk_decompressor.decompress_remaining_extend(&mut res)?;
```

Comment:
IIUC, this line (via [this](https://github.com/pcodec/pcodec/blob/c4c747784440ff05b9ab3a9a17fdaa5a821bbdf0/pco/src/standalone/decompressor.rs#L233) line, via [`Vec::reserve`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.reserve)) is where maybe-reallocation happens.  (It needed to happen somewhere, per the bug described in PR description.)

Reply:
Tangentially, can these 6 lines be replaced with a call to `Vec::resize_with`?
https://github.com/pcodec/pcodec/blob/c4c747784440ff05b9ab3a9a17fdaa5a821bbdf0/pco/src/standalone/decompressor.rs#L231-L236

Reply:
Unfortunately not, since we want to leave the memory uninitialized for performance reasons. We have to do something unsafe here. It would be nice if Rust had an unsafe one-liner for this, but I don't see such a thing.